sentry_sdk/profiler.py

```diff
@@ -203,20 +195,42 @@ def to_json(self):
         assert self._start_ns is not None
         assert self._stop_ns is not None
 
+        relative_start_ns = self._start_ns - int(
+            self.transaction._start_timestamp_monotonic * 1e9
+        )
+        relative_end_ns = nanosecond_time() - int(
+            self.transaction._start_timestamp_monotonic * 1e9
+        )
+
         return {
-            "device_os_name": platform.system(),
-            "device_os_version": platform.release(),
-            "duration_ns": str(self._stop_ns - self._start_ns),
             "environment": None,  # Gets added in client.py
+            "event_id": uuid.uuid4().hex,
             "platform": "python",
-            "platform_version": platform.python_version(),
-            "profile_id": uuid.uuid4().hex,
             "profile": _sample_buffer.slice_profile(self._start_ns, self._stop_ns),
-            "trace_id": self.transaction.trace_id,
-            "transaction_id": None,  # Gets added in client.py
-            "transaction_name": self.transaction.name,
-            "version_code": "",  # TODO: Determine appropriate value. Currently set to empty string so profile will not get rejected.
-            "version_name": None,  # Gets added in client.py
+            "release": None,  # Gets added in client.py
+            "timestamp": None,  # Gets added in client.py
+            "version": "1",
+            "device": {
+                "architecture": platform.machine(),
+            },
+            "os": {
+                "name": platform.system(),
+                "version": platform.release(),
+            },
+            "runtime": {
+                "name": platform.python_implementation(),
+                "version": platform.python_version(),
+            },
+            "transactions": [
+                {
+                    "id": None,  # Gets added in client.py
+                    "name": self.transaction.name,
+                    "relative_start_ns": str(relative_start_ns),
+                    "relative_end_ns": str(relative_end_ns),
+                    "trace_id": self.transaction.trace_id,
+                    "thread_id": str(self.transaction._thread_id),
```

Comment:
I don't think you need to get the `_thread_id` on the transaction. Let's move it to the profile's `__init__` and that should suffice. Just trying to keep profiling code separated from transactions where possible.